### PR TITLE
XRootD: add checksum + patch for 5.5.1

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

**Review by tldahlgren — DISMISSED** [review]
Confirmed the new sha256.  (FWIW Was unable to fetch/download the package due to a time out.)

File: var/spack/repos/builtin/packages/xrootd/package.py [hunk 1]
```diff
@@ -15,6 +15,7 @@ class Xrootd(CMakePackage):
     url = "http://xrootd.org/download/v5.3.1/xrootd-5.3.1.tar.gz"
     list_url = "https://xrootd.slac.stanford.edu/dload.html"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/xrootd/package.py]: We generally prefer that fetch (of a version) use the `https` protocol for downloads.  You could change the `url` ~and remove the `list_url`~ as follows and all of the checksums are still valid:

```suggestion
    url = "https://xrootd.slac.stanford.edu/download/v5.5.1/xrootd-5.5.1.tar.gz"
```

**Comment by iarspider** [comment]
@tldahlgren out of curiosity - why did you remove `list_url`?

**Comment by tldahlgren** [comment]
> @tldahlgren out of curiosity - why did you remove `list_url`?

Apologies.  I should've left it in for 'spack checksum rooted' to list the package versions.

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**